### PR TITLE
Rename idl to udl

### PR DESCRIPTION
**Pull request by eoger**
Fixes https://github.com/mozilla/uniffi-rs/issues/293.

This is a big breaking change, but it's worth doing before uniffi deploys in Mozilla's projects.

### CODE REVIEW

File: docs/manual/src/Overview.md [hunk 1]
```diff
@@ -6,8 +6,8 @@ Note that this tool will not help you ship a Rust library to these platforms, bu
 
 ## Design
 
-uniffi requires to write an Interface Definition Language ([IDL](https://heycam.github.io/webidl/)) file describing the methods and data structures available to the targeted languages.  
-This .idl file, whose definitions must match with the exposed Rust code, is then used to generate Rust *scaffolding* code and foreign-languages *bindings*. This process can take place either during the build process or be manually initiated by the developer.
+uniffi requires to write an Interface Definition Language (loosely based on [WebIDL](https://heycam.github.io/webidl/)) file describing the methods and data structures available to the targeted languages.
```
Comment (jhugman) [inline docs/manual/src/Overview.md:9]: Syntactically the same, but with different standard types and annotations.

File: README.md [hunk 2]
```diff
@@ -46,11 +46,11 @@ continues to evolve.
 Using `uniffi`, you can:
 
 * Implement your software component as a `cdylib` crate in Rust; let's say the code is in `./src/lib.rs`.
-* Specify the desired component API using an *Interface Definition Language* (specifically, a variant of WebIDL) in a separate file like `./src/lib.idl`.
-* Run `uniffi-bindgen scaffolding ./src/lib.idl` to generate a bunch of boilerplate rust code that exposes this API as a C-compatible FFI layer.
+* Specify the desired component API using an *Interface Definition Language* (specifically, a variant of WebIDL) in a separate file like `./src/lib.udl`.
```
Comment (jhugman) [inline README.md:49]: We don't say anywhere what `UDL` stands for.

**Review by jhugman — APPROVED** [review]
This looks good. 

Call out that we don't define what UDL stands for anywhere.

**Review by rfk — APPROVED** [review]
Thanks you for working through all these details!

**Pull request merged**